test/test_DeleteHandler.cpp

```diff
@@ -0,0 +1,100 @@
+#include "gmock/gmock.h"
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <stdexcept>
+#include <string>
+
+#include "MockFileSystemPolicy.hpp"
+
+#include "DeleteHandler.hpp"
+
+using ::testing::NiceMock;
+using ::testing::Return;
+
+class DeleteHandlerTest : public ::testing::Test {
+protected:
+	DeleteHandlerTest()
+	{
+		ON_CALL(m_fileSystemPolicy, deleteFile).WillByDefault(Return());
+	}
+	~DeleteHandlerTest() override { }
+
+	std::string m_path = "/workspaces/webserv/html/test/";
+	statusCode m_statusCode = StatusOK;
+
+	NiceMock<MockFileSystemPolicy> m_fileSystemPolicy;
+	DeleteHandler m_deleteHandler = DeleteHandler(m_fileSystemPolicy);
+};
+
+TEST_F(DeleteHandlerTest, DeleteFile)
+{
+	// Arrange
+	EXPECT_CALL(m_fileSystemPolicy, checkFileType).WillOnce(Return(FileSystemPolicy::FileRegular));
+
+	// Act
+	std::string responseBody = m_deleteHandler.execute(m_path, m_statusCode);
+
+	// Assert
+	EXPECT_EQ(responseBody,
+		"{\n\"message\": \"File deleted successfully\",\n\"file\": \"" + m_path
+			+ "\"\n}\n");
+}
+
+TEST_F(DeleteHandlerTest, DeleteDirectory)
```

Comment:
Need to comment out since we cant use deleteDirectory()

Reply:
Comment introduced